pandatool/src/deploy-stub/deploy-stub.c

```diff
@@ -628,6 +626,62 @@ static void unmap_blob(void *blob) {
   }
 }
 
+/**
+ * Gets the file path to the games .bin file which stores the blob made by FreezeTool
+ */
+static void get_blob_path(const char *path, char buffer[]) {
+#ifdef MACOS_APP_BUNDLE
+    uint32_t bufsize = PATH_MAX;
+    
+    if (_NSGetExecutablePath(buffer, &bufsize) != 0) {
+        exit(1);
+    }
+    char resolved[PATH_MAX];
+    
+    if (!realpath(buffer, resolved)) {
+        exit(1);
+    }
+    const char *dir = dirname(resolved);
+    sprintf(buffer, "%s/../Resources/%s.bin", dir, basename(resolved));
+#elif defined(MS_WINDOWS)
+    char buff[MAX_PATH];
+    DWORD length = GetModuleFileName(NULL, buff, MAX_PATH);
+    if (length == 0){
+        exit(1);
+    }
+    char fname[_MAX_FNAME];
+    _splitpath(buff, NULL, NULL, fname, NULL);
+    sprintf(buffer, "%s.bin", fname);
+#else
+    sprintf(buffer, "%s.bin", basename(path));
+#endif
+}
+
+/*
+ * When we are dealing with an external file, we have to load
+ * the blobinfo manually. This is to resolve an codesigning issues
+ * on ARM64 MacOS
+ */
+static void *map_external_blob(const char *path) {
+    FILE *runtime;
```

Comment:
We should call this variable something else. We're not opening the runtime file, so this is confusing. Maybe blob_file?